types/cloudevents-sdk/index.d.ts

```diff
@@ -0,0 +1,40 @@
+// Type definitions for cloudevents-sdk 0.3
+// Project: https://github.com/cloudevents/sdk-javascript (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
+// Definitions by: Andy Cunningham <https://github.com/andycmaj>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// tslint:disable-next-line no-declare-current-package no-single-declare-module cloudevents-sdk/v03 is the documented way to import the cloudevent builder
+declare module 'cloudevents-sdk/v03' {
+    // Cloudevent fluent builder
+    // https://github.com/cloudevents/sdk-javascript/blob/master/lib/cloudevent.js
+    interface CloudEventBuilder {
+        type: (type: string) => CloudEventBuilder;
+        dataContentType: (contentType: string) => CloudEventBuilder;
+        id: (id: string) => CloudEventBuilder;
+        time: (timestamp: Date) => CloudEventBuilder;
+        source: (source: string) => CloudEventBuilder;
+        data: (data: any) => CloudEventBuilder;
+
+        format: () => CloudEvent;
+        toString: () => string;
+    }
+
+    // Cloudevent.event() static constructor
+    // https://github.com/cloudevents/sdk-javascript#an-easy-way-to-create-events
+    const Cloudevent: {
+        event: () => CloudEventBuilder;
+    };
+
+    export default Cloudevent;
```

Comment:
[This is not how the package is exported](https://github.com/cloudevents/sdk-javascript/blob/0ac40bc5b0e9c68b0686cff9973cf6501dea8d62/v03/index.js#L19-L27), as it does not use `exports.default = `.

The best way to define exports for this package is to replace L24-L28 with just the following:

```ts
    function event(): CloudEventBuilder;
```

This way, the `CloudEventBuilder` interface (and `CloudEvent` interface once it is moved into the module), can all be imported via `import { } from 'cloudevents-sdk/v03';`.

Reply:
ah so the idea is to mirror the exact import you'd have to use when using the module's exports. that way the types become available with the imports you're already using? 

is that the idea?

Reply:
so typescript compilation failed with:

```bash
Error in cloudevents-sdk
Error: /Workspace/github/DefinitelyTyped/types/cloudevents-sdk/cloudevents-sdk-tests.ts:1:8
ERROR: 1:8  expect  TypeScript@next compile error:
Module '"cloudevents-sdk/v03"' has no default export.

    at /Workspace/github/DefinitelyTyped/node_modules/dtslint/bin/index.js:190:19
    at Generator.next (<anonymous>)
    at fulfilled (/Workspace/github/DefinitelyTyped/node_modules/dtslint/bin/index.js:5:58)
Error: The following packages had errors: cloudevents-sdk
    at doRunTests (/Workspace/github/DefinitelyTyped/node_modules/types-publisher/src/tester/test-runner.ts:240:11)
    at process._tickCallback (internal/process/next_tick.js:68:7)
npm ERR! Test failed.  See above for more details.
```

But as i understand, using `allowSyntheticDefaultImports` 'fixes' this warning. would you recommend using that in my `tsconfig.json`? There DO seem to be a number of other DT packages that are using this compiler option... so i've submitted that change here.